app/Models/DeviceRelatedModel.php

```diff
@@ -36,8 +36,9 @@ public function scopeHasAccess($query, User $user)
 
     public function scopeInDeviceGroup($query, $deviceGroup)
     {
-        $groups = $deviceGroup instanceof DeviceGroup ? $deviceGroup->devices()->pluck('devices.device_id') : [];
-        return $query->whereIn($this->getTable() . '.device_id', $groups);
+        return $query->whereHas('deviceGroup', function ($query) use ($deviceGroup) {
```

Comment:
@murrant what do you say about doing like this instead?
Will generate the SQL:
```sql
select * from `eventlog`
where exists (
  select * from `device_group_device`
  where `eventlog`.`device_id` = `device_group_device`.`device_id`
  and `device_group_device`.`device_group_id` = ?
)
``` 
That should be more efficient :)